modules/dmrpp_module/DmrppCommon.h

```diff
@@ -191,17 +207,32 @@ class DmrppCommon {
         return elements;
     }
 
+    /// @brief Set the uses_fill_value property
+    virtual void set_uses_fill_value(bool ufv) { d_uses_fill_value = ufv; }
+
+    /// @brief Set the fill value (using a string)
+    virtual void set_fill_value_string(const std::string &fv) { d_fill_value_str = fv; }
+
+    /// @brief Set the libdap data type to use with the fill value
+    virtual void set_fill_value_type(libdap::Type t) { d_fill_value_type = t; }
+
+    /// @return Return true if the the chunk uses 'fill value.'
+    virtual bool get_uses_fill_value() const { return d_uses_fill_value; }
+
+    /// @return Return the fill value as a string or "" if get_fill_value() is false
+    virtual std::string get_fill_value() const { return d_fill_value_str; }
+
+    /// @return Return the fill value as a string or "" if get_fill_value() is false
+    virtual libdap::Type get_fill_value_type() const { return d_fill_value_type; }
+
     void print_chunks_element(libdap::XMLWriter &xml, const std::string &name_space = "");
 
     void print_compact_element(libdap::XMLWriter &xml, const std::string &name_space = "", const std::string &encoded = "");
 
     void print_dmrpp(libdap::XMLWriter &writer, bool constrained = false);
 
-    // Replaced hsize_t with size_t. This eliminates a dependency on hdf5. jhrg 9/7/18
     /// @brief Set the value of the chunk dimension sizes given a vector of HDF5 hsize_t
-    void set_chunk_dimension_sizes(const std::vector<size_t> &chunk_dims) {
-        // tried using copy(chunk_dims.begin(), chunk_dims.end(), d_chunk_dimension_sizes.begin())
-        // it didn't work, maybe because of the differing element types?
+    void set_chunk_dimension_sizes(const std::vector<unsigned long long> &chunk_dims) {
```

Comment:
Nice catch! That `size_t` business is trouble.

Reply:
Right. site_t is what vector::size() returns, so we need it, but we know our arrays can be bigger. We might switch from writing `unsigned long long` to `uint64_t` now that we have C++11 - it standardizes that while unsigned long long is not _guaranteed_ to be 64-bits.